### PR TITLE
[agw][mme] Remove unused MAX_TAI_SUPPORTED config restriction

### LINKED ISSUE #3405: [AGW][MME] Replace assertions with log message wherever possible
In general, MME should only assert (and crash) when it has hit a memory error. This task is to analyze all existing assertions and replacing them with log messages.

For example, an assertion like below can be replaces with log.

```
Oct 22 06:25:41 magma mme[8452]: Assertion ((((config_pP->served_tai.tac[i]) != (uint16_t) 0x0000) && ((config_pP->served_tai.tac[i]) != (uint16_t) 0xFFFE))) failed!
Oct 22 06:25:41 magma mme[8452]: In mme_config_parse_file() /home/vagrant/magma/lte/gateway/c/oai/tasks/mme_app/mme_config.c:679
Oct 22 06:25:41 magma mme[8452]: Invalid TAC value 0x0
Oct 22 06:25:41 magma mme[8452]:  Will Intentionaly raise SEGFAULT to be catched by GDB!
```

### LINKED ISSUE #5766: [agw] mme crashes when configuring more than 16 TAI
### Your Environment

- **Version: 1.3**
- **Affected Component:** AGW
- **Affected Subcomponent:** AGW
- **Deployment Environment:** bare-metal (AGW)

### Describe the Issue

When configuring enB with a number of TAI that exceed 16, AGW start crashing

mme.log

```
000000 Thu Mar 25 14:09:35 2021 7FB876CDB7C0 INFO  UTIL   common/log.c                    :0750    Initializing OAI logging Done
000001 Thu Mar 25 14:09:35 2021 7FB876CDB7C0 DEBUG CONFIG tasks/mme_app/mme_app_embedded_s:0067    mme_config.config_file /var/opt/magma/tmp/mme.conf
000002 Thu Mar 25 14:09:35 2021 7FB876CDB7C0 DEBUG CONFIG tasks/mme_app/mme_app_embedded_s:0104    spgw_config.config_file /var/opt/magma/tmp/spgw.conf
000003 Thu Mar 25 14:09:35 2021 7FB876CDB7C0 INFO  MME-AP tasks/mme_app/mme_config.c      :0617    Number of TAIs configured: 45
000004 Thu Mar 25 14:09:35 2021 7FB876CDB7C0 CRITI ASSERT tasks/mme_app/mme_config.c      :0624    Assertion (num <= 16) failed! Too many TAIs configured: 45 (Maximum supported: 16)
```

syslog

```
ar 25 14:09:35 magma systemd[1]: Started Magma OAI MME service.
Mar 25 14:09:35 magma mme[16444]: Initializing OAI Logging
Mar 25 14:09:35 magma mme[16444]: Initializing OAI Logging to syslog
Mar 25 14:09:35 magma mme[16444]: Initializing shared logging
Mar 25 14:09:35 magma mme[16444]: Initializing shared logging Done
Mar 25 14:09:35 magma mme[16444]: Initializing TIMER module
Mar 25 14:09:35 magma mme[16444]: Initializing TIMER module: DONE
Mar 25 14:09:35 magma mme[16444]: Assertion (num <= 16) failed!
Mar 25 14:09:35 magma mme[16444]: In mme_config_parse_file() /home/vagrant/magma/lte/gateway/c/oai/tasks/mme_app/mme_config.c:624
Mar 25 14:09:35 magma mme[16444]: Too many TAIs configured: 45 (Maximum supported: 16)
Mar 25 14:09:35 magma mme[16444]:  Will Intentionaly raise SEGFAULT to be catched by GDB!

```

**Expected behavior**
service shouldn't crash and number of TAI should be limited to 16

### PR DESCRIPTION
**Pull request by ssanadhya**
Signed-off-by: Shruti Sanadhya <ssanadhya@fb.com>

## Summary

MME was crashing if more than 16 TAIs are configured via NMS (#5766 ). The `MAX_TAI_SUPPORTED` value is only used while reading config, hence removing it.

Also replaced some AssertFatals with stderr logs, so as to avoid service crashes.

## Test Plan

Regression testing: `make integ_test`
Function testing: Tested on Spirent AGW with 20 eNBs, each with different TAC value, and confirmed that 
- MME does not go into crash loop
- All 20 eNBs are able to do S1 Setup
- Able to attach/detach 10 UEs per eNB (total 200 UEs)

### CODE REVIEW

File: lte/gateway/c/oai/tasks/mme_app/mme_config.c [hunk 1]
```diff
@@ -640,14 +640,11 @@ int mme_config_parse_file(mme_config_t* config_pP) {
...
-          num >= MIN_TAI_SUPPORTED,
-          "Not even one TAI is configured, configure minimum one TAI\n");
-      AssertFatal(
-          num <= MAX_TAI_SUPPORTED,
-          "Too many TAIs configured: %d (Maximum supported: %d)", num,
-          MAX_TAI_SUPPORTED);
+      if (num <= MIN_TAI_SUPPORTED) {
+        fprintf(
+            stderr,
+            "Not even one TAI is configured, configure minimum one TAI\n");
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/mme_app/mme_config.c]: Maybe reword as "No TAI is configured. At least one TAI must be configured.\n"

**Review by ulaskozat — APPROVED** [review]
Just a nit is left.

File: lte/gateway/c/oai/tasks/mme_app/mme_config.c (hunk 1 above)
Comment (ssanadhya) [reply lte/gateway/c/oai/tasks/mme_app/mme_config.c]: Fixed in latest version

**Review by ardzoht — APPROVED** [review]
lgtm

**Comment by ssanadhya** [comment]
@rdefosse , could you please comment on the failure of https://jenkins-oai.eurecom.fr/blue/organizations/jenkins/MAGMA-MME-production/detail/MAGMA-MME-production/720/pipeline ?
